app/code/Magento/Catalog/Model/CompareListIdToMaskedListId.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Catalog\Model;
+
+use Magento\Catalog\Model\ResourceModel\Product\Compare\ListIdMask as ListIdMaskMaskResource;
+
+/**
+ *  CompareListId to MaskedListId resolver
+ */
+class CompareListIdToMaskedListId
+{
+    /**
+     * @var ListIdMaskFactory
+     */
+    private $listIdMaskFactory;
+
+    /**
+     * @var ListIdMaskMaskResource
+     */
+    private $listIdMaskResource;
+
+    /**
+     * @param ListIdMaskFactory $listIdMaskFactory
+     * @param ListIdMaskMaskResource $listIdMaskResource
+     */
+    public function __construct(
+        ListIdMaskFactory $listIdMaskFactory,
+        ListIdMaskMaskResource $listIdMaskResource
+    ) {
+        $this->listIdMaskFactory = $listIdMaskFactory;
+        $this->listIdMaskResource = $listIdMaskResource;
+    }
+
+    /**
+     * Get maskedId by listId
+     *
+     * @param int $listId
+     *
+     * @return string
+     */
+    public function execute(int $listId): string
+    {
+        $listIdMask = $this->listIdMaskFactory->create();
+        $this->listIdMaskResource->load($listIdMask, $listId, 'list_id');
+        return $listIdMask->getMaskedId() ?? '';
```

Comment:
I would suggest returning `null` instead of empty string. In that way you will make behaviour of this method more predictable

Reply:
**Done**